distributed/scheduler.py

```diff
@@ -2052,6 +2052,7 @@ def decide_worker_rootish_queuing_disabled(
             and tg.last_worker_tasks_left
             and lws.status == Status.running
             and self.workers.get(lws.address) is lws
+            and len(tg) > self.total_nthreads * 2
```

Comment:
Putting this statement here makes it also much easier to motivate it.

Below, we're calculating the number of tasks of the given task group we can assign safely to a worker `(len(tg) / self.total_nthreads) * ws.nthreads`

Therefore, `last_worker_tasks_left > 2 * ws.nthreads ~ tasks_assigned / thread > 2`

This means, we're only scheduling using co-assignment if we can at least assign two tasks per thread. Otherwise, we abort to protect our ability to scale out and use the ordinary worker_objective to distribute tasks

Reply:
Whether or not this "at least two tasks per thread" logic is a good choice is an entirely different matter, of course. 
However, this gives some meaning to this otherwise arbitrary choice